contracts/base/escrow/dapps/Aave.sol

```diff
@@ -0,0 +1,135 @@
+/**  itself upgradeable, they cannot have their own storage variables as they would cause the the   **/
+/**  storage slots to be overwritten on the Escrow proxy contract!                                  **/
+/**                                                                                                 **/
+/**  Visit https://docs.openzeppelin.com/upgrades/2.6/proxies#upgrading-via-the-proxy-pattern for   **/
+/**  more information.                                                                              **/
+/*****************************************************************************************************/
+/**
+    @notice This contract is used to define Aave dApp actions available. All dapp actions are invoked via
+        delegatecalls from Escrow contract, so this contract's state is really Escrow.
+    @author develop@teller.finance
+ */
+contract Aave is IAave, BaseEscrowDapp {
+    using AddressLib for address;
+    using Address for address;
+    using SafeERC20 for IERC20;
+
+    /**
+        @notice The deposit, the aToken needs to be approved to have access to the token balance
+        @param tokenAddress address of the token
+        @param amount amount of tokens to deposit
+     */
+    function deposit(address tokenAddress, uint256 amount) public onlyBorrower {
+        IAToken aToken = _getAToken(tokenAddress);
+        IAaveLendingPool aaveLendingPool = _getAavePoolAddress(tokenAddress);
+        uint256 aTokenBalanceBeforeDeposit = aToken.balanceOf(address(this));
+        IERC20(tokenAddress).safeApprove(address(aaveLendingPool), amount);
+        aaveLendingPool.deposit(tokenAddress, amount, address(this), 0);
+
+        uint256 aTokenBalanceAfterDeposit = aToken.balanceOf(address(this));
+        require(
+            aTokenBalanceAfterDeposit > aTokenBalanceBeforeDeposit,
+            "AAVE_BALANCE_NOT_INCREASED"
+        );
+
+        _tokenUpdated(address(aToken));
+        _tokenUpdated(tokenAddress);
+
+        uint256 tokenBalanceAfterDeposit = _balanceOf(tokenAddress);
+        emit AaveDeposited(
+            tokenAddress,
+            address(aToken),
+            amount,
+            tokenBalanceAfterDeposit,
+            aTokenBalanceAfterDeposit
+        );
+    }
+
+    /**
+        @notice This function withdraws the user's aTokens for a specific amount
+        @param tokenAddress address of the token
+        @param amount amount of the underlying tokens to withdraw
+     */
+    function withdraw(address tokenAddress, uint256 amount)
+        public
+        onlyBorrower
+    {
+        IAToken aToken = _getAToken(tokenAddress);
+        IAaveLendingPool aaveLendingPool = _getAavePoolAddress(tokenAddress);
+        uint256 aTokenBalanceBeforeWithdraw = aToken.balanceOf(address(this));
+        require(aTokenBalanceBeforeWithdraw >= 0, "NO_BALANCE_TO_WITHDRAW");
+        uint256 result =
+            aaveLendingPool.withdraw(tokenAddress, amount, address(this));
```

Comment:
Is there a value we should be checking for?

Reply:
Added token balance check to the function